datafusion/src/physical_plan/sort_preserving_merge.rs

```diff
@@ -197,7 +197,7 @@ impl ExecutionPlan for SortPreservingMergeExec {
 /// `SortKeyCursor::compare` can then be used to compare the sort key pointed to
 /// by this row cursor, with that of another `SortKeyCursor`. A cursor stores
 /// a row comparator for each other cursor that it is compared to.
-struct SortKeyCursor<'a> {
+struct SortKeyCursor {
```

Comment:
I ported apache/arrow-rs#542 back to arrow2, please merge the latest master into your `ord` branch. @houqp